src/streamsync/core.py

```diff
@@ -1022,9 +1038,13 @@ def _call_handler_callable(self, event_type, target_component, instance_path, pa
                 arg_values.append(session_info)
 
         result = None
-        with contextlib.redirect_stdout(io.StringIO()) as f:
-            result = callable_handler(*arg_values)
-        captured_stdout = f.getvalue()
+        if is_async_handler:
+            async_handler = self._async_handler_executor(callable_handler, arg_values)
+            thread_pool_future = thread_pool.submit(asyncio.run, async_handler)
```

Comment:
You are already in the thread pool executor, why calling the thread_pool again ? Is there a benefits of starting a new thread ?

![image](https://github.com/streamsync-cloud/streamsync/assets/159559/18f20125-bcfb-4721-ad12-b4bc8a15aaf8)

If you don't have to, it will avoid to move it through parameters. I join @ramedina86 on this remark of keeping the beast at the high level.

Reply:
I run your PR with this code on hello app

![image](https://github.com/streamsync-cloud/streamsync/assets/159559/0982bf7d-2d47-4ea8-8fa9-b97e7d9d2846)

Reply:
Originally this was a suggestion by @ramedina86 to use ThreadPoolExecutor here, and I imagine this could give us an advantage in terms of handling multiple calls. Your suggestion does work too indeed. We'd probably have to come up with some sort of benchmark test here, to better understand performance implications behind this. For now, let's indeed stick to this to keep things simpler parameter-passing-wise.